### PR TITLE
Fixes #13310 - Provide deep_clone and deep_clone!

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

File: app/models/job_invocation.rb [hunk 1]
```diff
@@ -93,12 +93,18 @@ def queued?
...
       invocation.description_format = self.description_format
       invocation.description = self.description
       invocation.pattern_template_invocations = self.pattern_template_invocations.map(&:deep_clone)
-      invocation.save!
+    end
+  end
+
+  def deep_clone!
+    deep_clone.tap do |clone|
+      clone.task_group.save!
```
Comment (stbenjam) [inline app/models/job_invocation.rb]: Wouldn't rails already automatically save this association?

Comment (adamruzicka) [inline app/models/job_invocation.rb]: Oh, it would. Good point

**Comment by stbenjam** [comment]
Looks good to me, ACK

**Comment by ares** [comment]
Thanks @adamruzicka, merging!

**Pull request merged**